src/components/TextInput/index.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { randomKey } from '../../utils/helpers'
+
+
+const TextInput = ({
+  name, label, onChange, value, errors, type,
+}) => (
+  <div className="input-field">
+    <input
+      type={type}
+      name={name}
+      id={name}
+      value={value}
+      className={`${errors.length > 0 ? ' invalid' : ''}`}
+      onChange={onChange}
+    />
+    <label htmlFor={name}>{label}</label>
+    {
+      errors.map(error => (
+        <span key={randomKey()} className="helper-text" data-error={error}/>
+      ))
+    }
+  </div>
+);
+
+TextInput.defaultProps = {
+  errors: [],
+};
+
+TextInput.propTypes = {
+  name: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired,
+  value: PropTypes.string.isRequired,
+  errors: PropTypes.arrayOf(PropTypes.string),
+  type: PropTypes.string.isRequired,
+};
+
+export const renderInputField = (onChange, value, label, errors, name, type = 'text') => (
```

Comment:
I dont see the need for this function, you can just export the component as it is and pass the props to it.

Reply:
I created this function to avoid repeating `onChange  method` every time when creating an input field. But I have refactored it.